Outline.org

```diff
@@ -0,0 +1,89 @@
+#+title: Modular Induction & Coinduction
+
+* Part the First
+
+** Reintroducing Recursion
+
+Here is just everything about recursion as you usually know it
+
+This chapter is probably mostly a review for you. But it serves to get us all on
+the same page with concepts and terminology. Unfortunately, this book also
+involves "unlearning" much of this. Perhaps some day there we can write a book
+that introduces the modularity described in this book directly, rather than
+having to de-program everyone.
+
+So, what /is/ recursion? 
+
+As an industry, we often contrast recursion with iteration, but recursion subsumes
+iteration. Syntactically, many languages have a special syntax for iteration
+constructs, but there is nothing that prevents us from implementing that syntax
+with recursion.
+
+So, what are some recursive (or iterative) things  you can think of?
+
+- applying some function to every element in a list
+- 
+
+*** Exercises
+
+- mostly simple, at the end of each section
+- last section introduces the "project"
+
+** Generalized traversals
+
+Uniplate style, introduces traversals without yet getting into algebras and pattern functors.
+
+** Guaranteed structure
+
+Uniplate lets functions dig arbitrarily deep into a tree, and they pair each
+element with a list of child results (~para~?), so it's quite possible that the
+list of results doesn't correspond to the children of the current node. How do
+we fix this?
+
+This section probably also /separates/ recursion from the business logic.
+
+** total traversals
+
+This section maybe eliminates recursion altogether
+
+** little pieces
+
+- introduce modularity via `Cofree`
+
+- introduce fusion via `zygo`
+
+*** Examples
+
+Breaking down some of the more complicated examples using zygomorphism
+
+** top-down data passing
+
+This is the reason to not introduce unfolding earlier -- people often think of
```

Comment:
Something interesting to note here: people are actually used to this style by passing an "accumulator". Another way this sounds familiar is difference lists.